test: 'auto' not allowed in function prototype with clang

https://github.com/pingcap/tics/blob/7d517dfa6f68f1180f788c568a3c5ee7adcc0805/dbms/src/Encryption/tests/gtest_rate_limiter.cpp#L522-L528

It's okay on GCC, but on clang 12.0:

```
../dbms/src/Encryption/tests/gtest_rate_limiter.cpp:522:50: error: 'auto' not allowed in function prototype
IOLimitTuner::Watermark watermarkOfBgWrite(const auto & tuner) { return tuner.getWatermark(tuner.bg_write_stat->pct()); }
                                                 ^~~~
../dbms/src/Encryption/tests/gtest_rate_limiter.cpp:524:50: error: 'auto' not allowed in function prototype
IOLimitTuner::Watermark watermarkOfFgWrite(const auto & tuner) { return tuner.getWatermark(tuner.fg_write_stat->pct()); }
                                                 ^~~~
../dbms/src/Encryption/tests/gtest_rate_limiter.cpp:526:49: error: 'auto' not allowed in function prototype
IOLimitTuner::Watermark watermarkOfBgRead(const auto & tuner) { return tuner.getWatermark(tuner.bg_read_stat->pct()); }
                                                ^~~~
../dbms/src/Encryption/tests/gtest_rate_limiter.cpp:528:49: error: 'auto' not allowed in function prototype
IOLimitTuner::Watermark watermarkOfFgRead(const auto & tuner) { return tuner.getWatermark(tuner.fg_read_stat->pct()); }
                                                ^~~~
4 errors generated.
```

Related StackOverflow thread: https://stackoverflow.com/questions/30665506/auto-not-allowed-in-function-prototype-with-clang

--- a/dbms/src/Encryption/tests/gtest_rate_limiter.cpp
+++ b/dbms/src/Encryption/tests/gtest_rate_limiter.cpp
@@ -519,13 +519,17 @@ TEST(IOLimitTuner_test, NotNeedTune)
     }
 }
 
-IOLimitTuner::Watermark watermarkOfBgWrite(const auto & tuner) { return tuner.getWatermark(tuner.bg_write_stat->pct()); }
+template <typename T>
+IOLimitTuner::Watermark watermarkOfBgWrite(const T & tuner) { return tuner.getWatermark(tuner.bg_write_stat->pct()); }
 
-IOLimitTuner::Watermark watermarkOfFgWrite(const auto & tuner) { return tuner.getWatermark(tuner.fg_write_stat->pct()); }
+template <typename T>
+IOLimitTuner::Watermark watermarkOfFgWrite(const T & tuner) { return tuner.getWatermark(tuner.fg_write_stat->pct()); }
 
-IOLimitTuner::Watermark watermarkOfBgRead(const auto & tuner) { return tuner.getWatermark(tuner.bg_read_stat->pct()); }
+template <typename T>
+IOLimitTuner::Watermark watermarkOfBgRead(const T & tuner) { return tuner.getWatermark(tuner.bg_read_stat->pct()); }
 
-IOLimitTuner::Watermark watermarkOfFgRead(const auto & tuner) { return tuner.getWatermark(tuner.fg_read_stat->pct()); }
+template <typename T>
+IOLimitTuner::Watermark watermarkOfFgRead(const T & tuner) { return tuner.getWatermark(tuner.fg_read_stat->pct()); }
 
 void updateWatermarkPct(StorageIORateLimitConfig & io_config, int emergency, int high, int medium)
 {

